projects/generate_glitches/generate_glitches.py

```diff
@@ -0,0 +1,342 @@
+    times = triggers[:, time_col]
+    time_args = np.logical_and(times > start, times < stop)
+    triggers = triggers[time_args]
+
+    # apply snr thresh
+    day_snrs = triggers[:, snr_col]  # second column is snrs
+    snr_thresh_args = np.where(day_snrs > snr_thresh)
+    triggers = triggers[snr_thresh_args]
+
+    # re-set 'start' and 'stop' so we aren't querying unnecessary data
+    start = np.min(triggers[:, time_col]) - 2 * window
+    stop = np.max(triggers[:, time_col]) + 2 * window
+
+    logging.info(
+        f"Querying {stop - start} seconds of data for {len(triggers)} triggers"
+    )
+
+    # use gwdatafind to create frame cache
+    frames = gwdatafind.find_urls(
+        site=ifo.strip("1"),
+        frametype=f"{ifo}_{frame_type}",
+        gpsstart=int(start),
+        gpsend=int(stop),
+        urltype="file",
+        on_gaps="ignore",
+    )
+
+    # read frames all at once, then crop
+    # this time series for each trigger.
+    # Although its initially slower to read all the frames at once,
+    # I think this is overall faster than querying a 4096 frame for
+    # every trigger.
+
+    ts = TimeSeries.read(
+        frames, channel=f"{ifo}:{channel}", start=start, end=stop, pad=0
+    )
+
+    # for each trigger
+    for trigger in tqdm(triggers):
+        time = trigger[time_col]
+
+        try:
+            glitch_ts = ts.crop(time - window, time + window)
+
+            glitch_ts = glitch_ts.resample(sample_rate)
+
+            snrs.append(trigger[snr_col])
+            glitches.append(glitch_ts)
+
+        except ValueError:
+            logging.info(f"Data not available for trigger at time: {time}")
+            continue
+
+    glitches = np.array(glitches)
+    snrs = np.array(snrs)
+    return glitches, snrs
+
+
+@typeo
+def main(
+    snr_thresh: float,
+    start: float,
+    stop: float,
+    window: float,
+    omicron_dir: str,
+    out_dir: str,
+    channel: str,
+    frame_type: str,
+    sample_rate: float = 4096,
+    H1_veto_file: Optional[str] = None,
+    L1_veto_file: Optional[str] = None,
+):
+
+    """Simulates a set of glitches for both
+        H1 and L1 that can be added to background
+
+    Arguments:
+
+    - snr_thresh: snr threshold above which to keep as glitch
+    - start: start gpstime
+    - stop: stop gpstime
+    - window: half window around trigger time to query data for
+    - sample_rate: sampling frequency
+    - out_dir: output directory to which signals will be written
+    - omicron_dir: base directory of omicron triggers
+            (see /home/ethan.marx/bbhnet/generate-glitch-dataset/omicron/)
+    - channel: channel name used to read data
+    - frame_type: frame type for data discovery w/ gwdatafind
+    - H1_veto_file: path to file containing vetoes for H1
+    - L1_veto_file: path to file containing vetoes for L1
+    """
+
+    # create logging file in model_dir
+    logging.basicConfig(
+        filename=f"{out_dir}/log.log",
```

Comment:
Here and in a few other places we make some assumptions about path separators. I don't ever really anticipate using this script on Windows, but at the very least it's inconsistent with also using `os.path.join`. This is primarily an aesthetic issue but it might make sense to opt for either one or the other (using `"/"` explicitly everywhere or providing every path component as an argument to `os.path.join`), or opting for using `pathlib` as outlined above.